webgl_generator/webgl_generators/stdweb_gen.rs

```diff
@@ -0,0 +1,621 @@
+                match t_kind {
+                    &TypeKind::TypedArray(_) => Some(t),
+                    &TypeKind::Sequence(_) => None,
+                    _ => panic!("Union support is limited!")
+                }
+            }).next().expect("Union did not contain a TypedArray");
+
+            process_arg_type(t, registry, gc)
+        },
+        &TypeKind::Dictionary | &TypeKind::Interface => ProcessedArg::simple(format!("&{}", name)),
+        &TypeKind::Enum => ProcessedArg::simple(name),
+        &TypeKind::Typedef(ref t) => {
+            // We have to "look through" the typedef, as the correct parameter
+            // type is not representable using the alias.
+            assert!(t.optional);
+            process_arg_type(t, registry, gc)
+        },
+        &TypeKind::Any | &TypeKind::Object => {
+            let gp = gc.arg("T");
+            gc.constrain(format!("{}: JsSerialize", gp));
+            ProcessedArg::simple(gp)
+        },
+        &TypeKind::Callback(ref _args, ref _return_type) => {
+            let gp = gc.arg("F");
+            gc.constrain(format!("{}: FnOnce() + 'static", gp));
+            ProcessedArg { type_: gp, wrapper: ArgWrapper::Once, optional: false }
+        }
+    }
+}
+
+fn process_arg_type(type_: &Type, registry: &Registry, gc: &mut GenericContext) -> ProcessedArg {
+    let type_kind = registry.resolve_type(&type_.name);
+    let mut result = process_arg_type_kind(&type_.name, &type_kind, registry, gc);
+    if type_.optional && !result.optional {
+        result.type_ = format!("Option<{}>", result.type_);
+        result.wrapper = match result.wrapper {
+            ArgWrapper::None => ArgWrapper::None,
+            other => ArgWrapper::Optional(Box::new(other))
+        };
+        result.optional = true;
+    }
+    result
+}
+
+#[derive(Clone, Debug)]
+enum ResultWrapper {
+    TryInto,
+    Ok
+}
+
+impl ResultWrapper {
+    fn wrap(&self, content: &str) -> String {
+        match self {
+            &ResultWrapper::TryInto => format!("{}.try_into().unwrap()", content),
+            &ResultWrapper::Ok => format!("{}.try_into().ok()", content),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+struct ProcessedResult {
+    type_: String,
+    wrapper: ResultWrapper,
+    optional: bool,
+}
+
+impl ProcessedResult {
+    fn simple<S: Into<String>>(name: S) -> ProcessedResult {
+        ProcessedResult { type_: name.into(), wrapper: ResultWrapper::TryInto, optional: false }
+    }
+}
+
+fn process_result_type_kind(name: &str, type_kind: &TypeKind, registry: &Registry) -> ProcessedResult {
+    match type_kind {
+        &TypeKind::Primitive(ref p) => ProcessedResult::simple(p.name()),
+        &TypeKind::String => ProcessedResult::simple("String"),
+        &TypeKind::ArrayBuffer | &TypeKind::ArrayBufferView => ProcessedResult::simple("ArrayBuffer"),
+        &TypeKind::BufferSource => unimplemented!("BufferSource not supported in output"),
```

Comment:
Should we be using `unimplemented!` here? Are you planning to add support for this later?

Reply:
`BufferSource` is not currently used in the WebGL IDL files, so I haven't bothered implementing it.